api/spec/spec_handler.go

```diff
@@ -309,12 +309,14 @@ func (d *specHandler) UpdateSpecFromOpts(opts map[string]string, spec *api.Volum
 		case api.SpecSecure:
 			if secure, err := strconv.ParseBool(v); err != nil {
 				return nil, nil, nil, err
-			} else {
+			} else if !spec.IsPureVolume() {
 				spec.Encrypted = secure
 			}
 		case api.SpecPassphrase:
-			spec.Encrypted = true
-			spec.Passphrase = v
+			if !spec.IsPureVolume() {
```

Comment:
Why do we need to do this? I would suggest instead let this module set all the VolumeSpec params as passed in opts. 
The PureVolume filter should return a proper error when it sees conflicting params set. With this change the user will be able to create a volume but we won't error out and the user might think they created a Pure volume with encryption set.